examples/descr_locations.rs

```diff
@@ -0,0 +1,107 @@
+use std::panic::Location;
+
+use dptree::{di::DependencyMap, Handler, HandlerDescription};
+
+struct CollectLocations {
+    locations: Vec<(&'static str, &'static Location<'static>)>,
+}
+
+#[track_caller]
+fn collector(name: &'static str) -> CollectLocations {
+    CollectLocations { locations: vec![(name, Location::caller())] }
+}
+
+impl HandlerDescription for CollectLocations {
+    #[track_caller]
+    fn entry() -> Self {
+        collector("entry")
+    }
+
+    #[track_caller]
+    fn user_defined() -> Self {
+        collector("user_defined")
+    }
+
+    #[track_caller]
+    fn merge_chain(&self, other: &Self) -> Self {
+        let locations = self
+            .locations
+            .iter()
+            .chain(&other.locations)
+            .copied()
+            .chain([("chain", Location::caller())])
+            .collect();
+
+        Self { locations }
+    }
+
+    #[track_caller]
+    fn merge_branch(&self, other: &Self) -> Self {
+        let locations = self
+            .locations
+            .iter()
+            .chain(&other.locations)
+            .copied()
+            .chain([("branch", Location::caller())])
+            .collect();
+
+        Self { locations }
+    }
+
+    #[track_caller]
+    fn map() -> Self {
+        collector("map")
+    }
+
+    #[track_caller]
+    fn map_async() -> Self {
+        collector("map_async")
+    }
+
+    #[track_caller]
+    fn filter() -> Self {
+        collector("filter")
+    }
+
+    #[track_caller]
+    fn filter_async() -> Self {
+        collector("filter_async")
+    }
+
+    #[track_caller]
+    fn filter_map() -> Self {
+        collector("filter_map")
+    }
+
+    #[track_caller]
+    fn filter_map_async() -> Self {
+        collector("filter_map_async")
+    }
+
+    #[track_caller]
+    fn endpoint() -> Self {
+        collector("endpoint")
+    }
```

Comment:
Maybe generate these definitions via a macro? Something like `methods! { map, map_async, filter, filter_async, ... };` ?

Reply:
I don't think this is needed in this case, macro will only obscure the code